gobblin-metrics/src/main/java/gobblin/metrics/kafka/KafkaReporter.java

```diff
@@ -61,10 +67,15 @@
   private final String topic;
   private final ObjectMapper mapper = new ObjectMapper();
 
+  private final SpecificDatumWriter<MetricReport> writer;
+  private final Encoder encoder;
+  private final ByteArrayOutputStream out;
+
   protected long lastSerializeExceptionTime;
   protected final Closer closer;
 
   public final Map<String, String> tags;
+  private boolean reportMetrics = true;
```

Comment:
You can make this final and set the value based on the result of `getEncoder`. So if `getEncoder` returns `null`, then `reportMetrics` is `false`, otherwise it is `true`.